core/src/main/java/io/questdb/griffin/model/ExpressionNode.java

```diff
@@ -167,15 +167,19 @@ public boolean equals(Object o) {
                 && Objects.equals(rhs, that.rhs);
     }
 
-    public boolean hasLeafs() {
-        return lhs != null && rhs != null;
+    public boolean noLeafs() {
```

Comment:
why not call this method `x2`? 

not to have leafs == left leaf is null and right leaf is null.
in the condition only one null is required, so a better name would be 'maybeOneLeaf?'

Reply:
this was always reversed, e.g. used in `!hasLeafs()`, intellij kept complaining, so i reversed the original method in easiest way possible to stop the complaints